File: Javascript/JavaScript Tasks/1/src/expression/Or.java
package expression;

import expression.exceptions.BinaryOperation;

public class Or extends BinaryOperation {

    public Or(TripleExpression left, TripleExpression right) {
        super(left, right);
    }

    public int eval(int l, int r) {
        return l | r;
    }

    public double eval(double l, double r) {
        return (int) l | (int) r;
    }
}

File: Mathlogic/B-D/src/expression/Or.kt
package expression

class Or(left : Expression, right: Expression) : BinaryOperator(left, right, "|") {
    override fun getNewExpr(letterToExpr: Map<String, Expression>): Expression {
        return Or(left.getNewExpr(letterToExpr),right.getNewExpr(letterToExpr))
    }
}